hack/build_release_image.sh

```diff
@@ -0,0 +1,141 @@
+    echo "Options:"
+    echo "-h, --help      show this message"
+    echo "-u, --username  registered username in quay.io"    
+    echo "-t, --tag       push to a custom tag in your origin release image repo, default: latest"
+    echo "-r, --release   openshift release version, default: 4.8"
+    echo "-a, --auth      path of registry auth file, default: ./config.json"
+    echo "--aws-ccm       custom aws cloud-controller-manager image name"
+    echo "--openstack-ccm custom openstack cloud-controller-manager image name"
+    echo "--kapio         custom kube-apiserver-operator image name"
+    echo "--kcmo          custom kube-controller-manager-operator image name"
+    echo "--mco           custom machine-config-operator image name"
+}
+
+: ${GOPATH:=${HOME}/go}
+: ${TAG:="latest"}
+: ${RELEASE:="4.8"}
+: ${OC_REGISTRY_AUTH_FILE:="config.json"}
+: ${OCCM_IMAGE:="quay.io/openshift/origin-openstack-cloud-controller-manager:$RELEASE"}
+# TODO(mfedosin): replace upstream aws ccm image with "quay.io/openshift/origin-aws-cloud-controller-manager:$RELEASE"
+# when https://github.com/openshift/release/pull/18995 is merged
+: ${AWSCCM_IMAGE:="gcr.io/k8s-staging-provider-aws/cloud-controller-manager:v1.19.0-alpha.1"}
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h|--help)
+            help
+            exit 0
+            ;;
+            
+        -u|--username)
+            USERNAME=$2
+            shift 2
+            ;;
+
+        -t|--tag)
+            TAG=$2
+            shift 2
+            ;;
+
+        -r|--release)
+            RELEASE=$2
+            shift 2
+            ;;
+
+        -a|--auth)
+            OC_REGISTRY_AUTH_FILE=$2
+            shift 2
+            ;;
+
+        --aws-ccm)
+            AWSCCM_IMAGE=$2
+            shift 2
+            ;;
+
+        --openstack-ccm)
+            OCCM_IMAGE=$2
+            shift 2
+            ;;
+
+        --kapio)
+            KAPIO_IMAGE=$2
+            shift 2
+            ;;
+
+        --kcmo)
+            KCMO_IMAGE=$2
+            shift 2
+            ;;
+
+        --mco)
+            MCO_IMAGE=$2
+            shift 2
+            ;;
+
+        *)
+            echo "Invalid option $1"
+            help
+            exit 0
+            ;;
+    esac
+done
+
+if [ -z "$USERNAME" ]; then
+    echo "-u/--username was not provided, exiting ..."
+    exit 1
+fi
+
+if [ ! -f "$OC_REGISTRY_AUTH_FILE" ]; then
+    echo "$OC_REGISTRY_AUTH_FILE not found, exiting ..."
+    exit 1
+fi
+
+PREFIX="Pull From: "
+DEST_IMAGE="quay.io/$USERNAME/origin-release:$TAG"
+FROM_IMAGE=$(curl -s  https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp-dev-preview/latest-$RELEASE/release.txt | grep "$PREFIX" | sed -e "s/^$PREFIX//")
+CCCMO_IMAGE="quay.io/$USERNAME/cluster-cloud-controller-manager-operator:$TAG"
+
+echo "Start building CCCMO image $CCCMO_IMAGE"
+
+docker build --no-cache -t $CCCMO_IMAGE .
+docker push $CCCMO_IMAGE
```

Comment:
i wonder if we might put a flag around the push so that we could build locally without worrying about messing up the remote image?

i'd be ok with making the default to push and allowing to disable push with a flag, if that makes sense for this script.